src/main/ruby/truffleruby/core/truffle/polyglot.rb

```diff
@@ -34,6 +34,192 @@ def self.as_enumerable(object)
     Truffle::Interop.enumerable(object)
   end
 
+  # HasArrayElementsTrait
+  module HasArrayElementsTrait
+
+    def size
+      Truffle::Interop.array_size(self)
+    end
+
+    alias_method :length, :size
+
+    def count(item = nil)
+      seq = 0
+      if (!item.nil?)
+        each { |o| seq += 1 if item == o }
+      elsif block_given?
+        each { |o| seq += 1 if yield(o) }
+      else
+        return size
+      end
+      seq
+    end
+
+    def toArray
+      Truffle::Interop.invoke(self, :toArray)
+    end
+
+    def to_a
+      if Truffle::Interop.has_array_elements?(self)
+        Truffle::Interop.to_array(self)
+      else
+        raise RuntimeError.new
+      end
+    end
+
+    alias_method :to_ary, :to_a
+
+    def read_array_element(index)
+      Truffle::Interop.read_array_element(self, index)
+    end
+
+    def at(index)
+      index += length if index < 0
+      return nil if (index < 0 || index >= length)
+      read_array_element(index)
+    end
+
+    def first
+      read_array_element(0)
+    end
+
+    def last
+      read_array_element(length - 1)
+    end
+
+    def each(*args)
+      return enum_for(:each) unless block_given?
+      i = 0
+      while i < length
+        yield read_array_element(i)
+        i += 1
+      end
+    end
+
+    def each_with_index
+      return enum_for(:each) unless block_given?
+      i = 0
+      while i < length
+        yield read_array_element(i), i
+        i += 1
+      end
+    end
+
+  end
+
+  # ForeignObject class
+  class ForeignObject
+    # TODO: includes should be done dynamically depending on the type of the
+    # foreign object. Not sure yet how to do it
+    include Enumerable
+    include HasArrayElementsTrait
+
+    def getClass
+      Truffle::Interop.foreign_class(self)
+    end
+
+    alias_method :class, :getClass
+
+    def getName
+      Truffle::Interop.invoke(self, :getName)
+    end
+
+    def intValue
+      Truffle::Interop.invoke(self, :intValue)
+    end
+
+    def longValue
+      Truffle::Interop.invoke(self, :longValue)
+    end
```

Comment:
What is this for?

Reply:
On file java_new_spec.rb there is the following spec:

big_integer.intValue.should == 14

I've created longValue and doubleValue just for symmetry.  Should all of them be removed and the spec file changed, or is intValue actually expected and the others are not?

Reply:
Same for getClass and getName.... used in the specs.

Reply:
That's a host object, all methods which are not found as Ruby method should be invoked as InvokeMemberNode:
https://github.com/oracle/truffleruby/pull/2153#discussion_r535357548